### PR TITLE
feat: Adds ability to edit out webhook URL of API Channel

### LINKED ISSUE #1878: feat: Ability to edit out going address of API Channel
At the moment on chatwoot, After creating an API channel, there isn't the ability to edit the webhook URL

### Acceptance: 

- [ ] As an admin, I should be able to come back and edit the webhook URL for an existing API channel
- [ ] Verify the event payloads are correctly delivered to the new webhook URL post update

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

Adds the ability to come back and edit out the webhook URL for an existing API channel

Fixes #1878 

## Type of change

**Screen Recording**

https://user-images.githubusercontent.com/64252451/133250019-1a4ab33b-a390-48d7-a749-4df16bd22f32.mov

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Can be tested by going to the inbox setting and select API inbox settings

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by nithindavid — APPROVED** [review]

**Comment by muhsin-k** [comment]
@iamsivin I have found 2 issues while testing 

1. Not able to edit the URL from the end of inout
2. Webbhook URL is a mandatory field

https://user-images.githubusercontent.com/12408980/134115152-4e04dc29-4509-4469-a593-4eddb3c16e5b.mov

**Comment by muhsin-k** [comment]
@iamsivin Still able to update the empty URL

https://user-images.githubusercontent.com/12408980/134274986-f4fbf28e-e449-4432-a6c9-d41bb546310e.mov

File: app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue [hunk 1]
```diff
@@ -334,6 +353,9 @@ import PreChatFormSettings from './PreChatForm/Settings';
 import WeeklyAvailability from './components/WeeklyAvailability';
 import GreetingsEditor from 'shared/components/GreetingsEditor';
 
+const shouldNoBeEmpty = (value = '') =>
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue:356]: Move this logic to validation helpers

Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue]: @muhsin-k can you help me in fixing the code climate error.

Comment (muhsin-k) [reply app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue]: @iamsivin Add spec for `shouldNoBeEmpty`  methods

File: app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue [hunk 2]
```diff
@@ -212,6 +230,14 @@
         </div>
 
         <woot-submit-button
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue:232]: We could use woot-button here with `type=""submit`

**Review by muhsin-k — DISMISSED** [review]

File: app/controllers/api/v1/accounts/inboxes_controller.rb [hunk 3]
```diff
@@ -102,6 +102,7 @@ def create_channel
   def update_channel_feature_flags
     return unless @inbox.web_widget?
     return unless permitted_params(Channel::WebWidget::EDITABLE_ATTRS)[:channel].key? :selected_feature_flags
+    return unless @inbox.channel.try(:selected_feature_flags)
```
[the remark this answers is not in the document]
Comment (sojan-official) [reply app/controllers/api/v1/accounts/inboxes_controller.rb:105]: should fix the code climate issue

**Pull request merged**